flaky unit test in pkg/client/unversioned/portforward

```
--- FAIL: TestForwardPorts (0.01s)
	portforward_test.go:270: forward 2 ports with bidirectional data: server expected to receive "ghij", got "" for port 6000
E1027 15:46:48.833324    1555 portforward.go:303] error copying from local connection to remote stream: write tcp 127.0.0.1:45208->127.0.0.1:59640: write: broken pipe
I1027 15:46:48.833625    1555 portforward.go:213] Forwarding from 127.0.0.1:5555 -> 5555
I1027 15:46:48.833663    1555 portforward.go:213] Forwarding from [::1]:5555 -> 5555
E1027 15:46:48.834143    1555 portforward.go:202] Unable to create listener: Error listen tcp4 127.0.0.1:5555: bind: address already in use
E1027 15:46:48.834190    1555 portforward.go:202] Unable to create listener: Error listen tcp6 [::1]:5555: bind: address already in use
W1027 15:46:48.834219    1555 portforward.go:154] Unable to listen on port 5555: All listeners failed to create with the following errors: listen tcp4 127.0.0.1:5555: bind: address already in use, listen tcp6 [::1]:5555: bind: address already in use
FAIL
FAIL	k8s.io/kubernetes/pkg/client/unversioned/portforward	0.188s
```

--- a/pkg/client/unversioned/portforward/portforward_test.go
+++ b/pkg/client/unversioned/portforward/portforward_test.go
@@ -207,6 +207,8 @@ func TestGetListener(t *testing.T) {
 // kubelet.PortForwarder.
 type fakePortForwarder struct {
 	lock sync.Mutex
+	// stores data expected from the stream per port
+	expected map[uint16]string
 	// stores data received from the stream per port
 	received map[uint16]string
 	// data to be sent to the stream per port
@@ -218,33 +220,23 @@ var _ kubelet.PortForwarder = &fakePortForwarder{}
 func (pf *fakePortForwarder) PortForward(name string, uid types.UID, port uint16, stream io.ReadWriteCloser) error {
 	defer stream.Close()
 
-	var wg sync.WaitGroup
+	// read from the client
+	received := make([]byte, len(pf.expected[port]))
+	n, err := stream.Read(received)
+	if err != nil {
+		return fmt.Errorf("error reading from client for port %d: %v", port, err)
+	}
+	if n != len(pf.expected[port]) {
+		return fmt.Errorf("unexpected length read from client for port %d: got %d, expected %d. data=%q", port, n, len(pf.expected[port]), string(received))
+	}
 
-	// client -> server
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+	// store the received content
+	pf.lock.Lock()
+	pf.received[port] = string(received)
+	pf.lock.Unlock()
 
-		// copy from stream into a buffer
-		received := new(bytes.Buffer)
-		io.Copy(received, stream)
-
-		// store the received content
-		pf.lock.Lock()
-		pf.received[port] = received.String()
-		pf.lock.Unlock()
-	}()
-
-	// server -> client
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-
-		// send the hardcoded data to the stream
-		io.Copy(stream, strings.NewReader(pf.send[port]))
-	}()
-
-	wg.Wait()
+	// send the hardcoded data to the client
+	io.Copy(stream, strings.NewReader(pf.send[port]))
 
 	return nil
 }
@@ -254,6 +246,7 @@ func (pf *fakePortForwarder) PortForward(name string, uid types.UID, port uint16
 func fakePortForwardServer(t *testing.T, testName string, serverSends, expectedFromClient map[uint16]string) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		pf := &fakePortForwarder{
+			expected: expectedFromClient,
 			received: make(map[uint16]string),
 			send:     serverSends,
 		}

